kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/system/SystemExitExtensions.kt

```diff
@@ -50,21 +51,30 @@ object SystemExitListener : TestListener {
  */
 object SpecSystemExitListener : TestListener {
 
-    private val previousSecurityManagers = ConcurrentHashMap<Description, SecurityManager>()
+   private val previousSecurityManagers = ConcurrentHashMap<Description, SecurityManager>()
 
    override suspend fun beforeSpec(spec: Spec) {
-        val previous = System.getSecurityManager()
-        if (previous != null)
-            previousSecurityManagers[spec::class.toDescription()] = previous
-        System.setSecurityManager(NoExitSecurityManager(previous))
-    }
+      val previous = System.getSecurityManager()
+      if (previous != null)
+         previousSecurityManagers[spec::class.toDescription()] = previous
+      System.setSecurityManager(NoExitSecurityManager(previous))
+   }
 
    override suspend fun afterSpec(spec: Spec) {
-        if (previousSecurityManagers.contains(spec::class.toDescription()))
-            System.setSecurityManager(previousSecurityManagers[spec::class.toDescription()])
-        else
-            System.setSecurityManager(null)
-    }
+      if (previousSecurityManagers.contains(spec::class.toDescription()))
+         System.setSecurityManager(previousSecurityManagers[spec::class.toDescription()])
+      else
+         System.setSecurityManager(null)
+   }
+
+   fun lastExitCode(): Int? = when (val manager = System.getSecurityManager()) {
+      is NoExitSecurityManager -> manager.lastExitCode
+      else -> null
+   }
```

Comment:
```suggestion
   fun lastExitCode() = (System.getSecurityManager() as? NoExitSecurityManager)?.lastExitCode
```
IDK if better, but more concise

Reply:
Harder to read imo